src/ol_infrastructure/applications/airbyte/__main__.py

```diff
@@ -0,0 +1,268 @@
+target_vpc = network_stack.require_output(target_vpc_name)
+
+consul_security_groups = consul_stack.require_output("security_groups")
+aws_config = AWSBase(
+    tags={
+        "OU": airbyte_config.get("business_unit") or "operations",
+        "Environment": f"{env_name}",
+    }
+)
+aws_account = get_caller_identity()
+vpc_id = target_vpc["id"]
+airbyte_server_ami = ec2.get_ami(
+    filters=[
+        ec2.GetAmiFilterArgs(name="name", values=["airbyte-server-*"]),
+        ec2.GetAmiFilterArgs(name="virtualization-type", values=["hvm"]),
+        ec2.GetAmiFilterArgs(name="root-device-type", values=["ebs"]),
+    ],
+    most_recent=True,
+    owners=[aws_account.account_id],
+)
+
+airbyte_server_tag = f"airbyte-server-{env_name}"
+consul_provider = get_consul_provider(stack_info)
+
+###############################
+##     General Resources     ##
+###############################
+
+# IAM and instance profile
+airbyte_server_instance_role = iam.Role(
+    f"airbyte-server-instance-role-{env_name}",
+    assume_role_policy=json.dumps(
+        {
+            "Version": "2012-10-17",
+            "Statement": {
+                "Effect": "Allow",
+                "Action": "sts:AssumeRole",
+                "Principal": {"Service": "ec2.amazonaws.com"},
+            },
+        }
+    ),
+    path="/ol-infrastructure/airbyte-server/role/",
+    tags=aws_config.tags,
+)
+iam.RolePolicyAttachment(
+    f"airbyte-server-describe-instance-role-policy-{env_name}",
+    policy_arn=policy_stack.require_output("iam_policies")["describe_instances"],
+    role=airbyte_server_instance_role.name,
+)
+iam.RolePolicyAttachment(
+    f"airbyte-server-route53-role-policy-{env_name}",
+    policy_arn=policy_stack.require_output("iam_policies")["route53_odl_zone_records"],
+    role=airbyte_server_instance_role.name,
+)
+airbyte_server_instance_profile = iam.InstanceProfile(
+    f"airbyte-server-instance-profile-{env_name}",
+    role=airbyte_server_instance_role.name,
+    path="/ol-infrastructure/airbyte-server/profile/",
+)
+
+# Vault policy definition
+airbyte_server_vault_policy = vault.Policy(
+    "airbyte-server-vault-policy",
+    name="airbyte-server",
+    policy=Path(__file__).parent.joinpath("airbyte_server_policy.hcl").read_text(),
+)
+# Register Tika AMI for Vault AWS auth
+vault.aws.AuthBackendRole(
+    "airbyte-server-ami-ec2-vault-auth",
+    backend="aws",
+    auth_type="iam",
+    role="airbyte-server",
+    inferred_entity_type="ec2_instance",
+    inferred_aws_region=aws_config.region,
+    bound_iam_instance_profile_arns=[airbyte_server_instance_profile.arn],
+    bound_ami_ids=[airbyte_server_ami.id],
+    bound_account_ids=[aws_account.account_id],
+    bound_vpc_ids=[vpc_id],
+    token_policies=[airbyte_server_vault_policy.name],
+)
+
+##################################
+#     Network Access Control     #
+##################################
+# Create security group
+airbyte_server_security_group = ec2.SecurityGroup(
```

Comment:
Unless we're using the sec group as an identifier for other ingress rules there's not much need for it, since every VPC has a web server group built into it https://github.com/mitodl/ol-infrastructure/blob/main/src/ol_infrastructure/infrastructure/aws/network/security_groups.py#L18-L55

Reply:
It's also being used as access control to the DB which is reason enough to keep it.